app/Jobs/ImportRockTheVotePosts.php

```diff
@@ -0,0 +1,459 @@
+            }
+
+            event(new LogProgress('', 'progress', ($offset / $this->totalRecords) * 100));
+        }
+
+        event(new LogProgress('Done!', 'general'));
+
+        Stat::create([
+            'filename' => $this->filepath,
+            'total_records' => $this->stats['totalRecords'],
+            'stats' => json_encode($this->stats),
+        ]);
+
+    }
+
+    /**
+     * Initiate the stat counters.
+     *
+     * @return array
+     */
+    private function statsInit()
+    {
+        return [
+            'totalRecords' => 0,
+            'countScrubbed' => 0,
+            'countProcessed' => 0,
+            'countPostCreated' => 0,
+            'countUserAccountsCreated' => 0,
+        ];
+    }
+
+    /**
+     * Parse the referral code field to grab individual values.
+     *
+     * @param  array $refferalCode
+     */
+    private function parseReferralCode($referralCode)
+    {
+        $values = [];
+
+        // Remove some nonsense that comes in front of the referral code sometimes
+        if (strrpos($referralCode, 'iframe?r=') !== false) {
+            $referralCode = str_replace('iframe?r=', null, $referralCode);  
+        }
+        if (strrpos($referralCode, 'iframe?') !== false) {
+            $referralCode = str_replace('iframe?', null, $referralCode);  
+        }
+
+        if (! empty($referralCode)) {
+            $referralCode = explode(',', $referralCode);
+
+            $referral = false;
+
+            foreach ($referralCode as $value) {
+
+                // See if we are dealing with ":" or "="
+                if (strpos($value, ':')) {
+                    $value = explode(':', $value);
+                }
+                elseif (strpos($value, '=')) {
+                    $value = explode('=', $value);
+                }
+
+                // Grab northstar id
+                if (strtolower($value[0]) === 'user') {
+                    $values['northstar_id'] = $value[1];
+                }
+
+                // Grab the Campaign Id.
+                if (strtolower($value[0]) === 'campaignid' || strtolower($value[0]) === 'campaign') {
+                    $values['campaign_id'] = $value[1];
+                }
+
+                // Grab the Campaign Run Id.
+                if (strtolower($value[0]) === 'campaignrunid') {
+                    $values['campaign_run_id'] = $value[1];
+                }
+
+                // Grab the source
+                if (strtolower($value[0]) === 'source') {
+                    $values['source'] = $value[1];
+                }
+
+                // Grab any source details
+                if (strtolower($value[0]) === 'source_details') {
+                    $values['source_details'] = $value[1];
+                }
+
+                // Is this a referral?
+                if (strtolower($value[0]) === 'referral' && strtolower($value[1]) === 'true') {
+                    $referral = true;
+                }
```

Comment:
I think this is totally preference but this could be a nice place to use a [switch](http://php.net/manual/en/control-structures.switch.php) statement?

Reply:
@chloealee yes I like this way better!!!!